### PR TITLE
[simdutf] Update to 3.2.16

### PR DESCRIPTION
**Pull request by JuliusBrueggemann**
# Changes in this PR
This PR updates the [simdutf](https://github.com/simdutf/simdutf) library to version 3.2.15 (from 1.0.0, which is 2 years old at this point). The important changes are:

- I removed the patch that disabled the tests and benchmarks since the new version allows disabling those with CMake options
- I added the "tools" feature, which builds the command line tool that was added by version 2.0.0 of `simdutf`

# Checklist
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Adela0814** [comment]
@JuliusBrueggemann  Please fix the ci error:
```
warning: The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    D:\packages\simdutf_x64-windows\bin\sutf.exe

warning: The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    D:\packages\simdutf_x64-windows\debug\bin\sutf.exe

error: Found 2 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: C:\a\1\s\ports\simdutf\portfile.cmake
error: building simdutf:x64-windows failed with: POST_BUILD_CHECKS_FAILED
```

File: ports/simdutf/portfile.cmake [hunk 1]
```diff
@@ -1,20 +1,27 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO simdutf/simdutf
-    REF v1.0.0
-    SHA512 3ab09aa61cb9765bf1a77de59c5b823ee58ae5e4badfd5dd70e511fd4f378f8d3917a5b577e7f275720b975740344968132ce0b3f628452bde67f2ab6cc82337
+    REF v3.2.15
```
Comment (Adela0814) [inline ports/simdutf/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

File: ports/simdutf/portfile.cmake [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
+)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    "tools" SIMDUTF_TOOLS
 )
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS "-DSIMDUTF_TESTS=OFF -DSIMDUTF_BENCHMARKS=OFF ${FEATURE_OPTIONS}"
```
Comment (Adela0814) [inline ports/simdutf/portfile.cmake:16]:
```suggestion
    OPTIONS ${FEATURE_OPTIONS}
        -DSIMDUTF_TESTS=OFF
        -DSIMDUTF_BENCHMARKS=OFF
```

File: ports/simdutf/portfile.cmake [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
 
 vcpkg_cmake_install()
 
 vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
+if ("tools" IN_LIST FEATURES)
+    vcpkg_copy_tools(TOOL_NAMES sutf AUTO_CLEAN)
+endif()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(INSTALL "${SOURCE_PATH}/LICENSE-APACHE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (Adela0814) [inline ports/simdutf/portfile.cmake:27]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE-APACHE")
```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by JuliusBrueggemann** [comment]
@Adela0814 I've applied your suggestions (which also fixed the CI error).

File: ports/simdutf/vcpkg.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
+  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
   "name": "simdutf",
-  "version-semver": "1.0.0",
+  "version-semver": "3.2.15",
   "port-version": 1,
```
Comment (Thomas1664) [inline ports/simdutf/vcpkg.json:5]:
```suggestion
  "version-semver": "3.2.15",
```

File: versions/s-/simdutf.json [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "c568ec805f427af56ad900604ee33917c191af2d",
+      "version-semver": "3.2.15",
+      "port-version": 1
+    },
+    {
+      "git-tree": "8e072bde394ff089f49dd567a68900f6b3cb3cea",
+      "version-semver": "3.2.15",
+      "port-version": 0
+    },
```
Comment (Thomas1664) [inline versions/s-/simdutf.json:12]: Please only add one version per PR. Revert your changes to this file, apply the suggestion above and run `vcpkg x-add-version` again. If you need to further update the version database, use `vcpkg x-add-version --overwrite-version` instead.

**Review by Thomas1664 — CHANGES_REQUESTED** [review]

File: ports/simdutf/vcpkg.json (hunk 4 above)
Comment (JuliusBrueggemann) [reply ports/simdutf/vcpkg.json]: :+1:

File: versions/s-/simdutf.json (hunk 5 above)
Comment (JuliusBrueggemann) [reply versions/s-/simdutf.json]: I had a feeling that what I was doing was wrong, I just didn't know what the right thing was. :laughing: I've reverted my changes to the version file, reset the port version to 0 and then reran `vcpkg x-add-version` like you requested.

Thanks for telling me about the `--overwrite-version` flag.

**Comment by Adela0814** [comment]
Feature testing failed on `x86-windows`:
repro step: `.\vcpkg.exe install simdutf[*]:x86-windows`
error:
```
E:\4\vcpkg\buildtrees\simdutf\src\v3.2.15-c99b139426.clean\tools\sutf.cpp(276): error C2220: the following warning is treated as an error
E:\4\vcpkg\buildtrees\simdutf\src\v3.2.15-c99b139426.clean\tools\sutf.cpp(138): note: see reference to function template instantiation 'void CommandLine::run_simdutf_procedure<CommandLine::run_procedure::<lambda_a55ba68c4042f8e6132c03ad63c272c3>>(PROCEDURE)' being compiled
        with
        [
            PROCEDURE=CommandLine::run_procedure::<lambda_a55ba68c4042f8e6132c03ad63c272c3>
        ]
E:\4\vcpkg\buildtrees\simdutf\src\v3.2.15-c99b139426.clean\tools\sutf.cpp(276): warning C4018: '<': signed/unsigned mismatch
```
details：
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/12289236/install-x64-windows-dbg-out.log)

**Comment by JuliusBrueggemann** [comment]
> Feature testing failed on `x86-windows`: repro step: `.\vcpkg.exe install simdutf[*]:x86-windows` error:
> 
> ```
> E:\4\vcpkg\buildtrees\simdutf\src\v3.2.15-c99b139426.clean\tools\sutf.cpp(276): error C2220: the following warning is treated as an error
> E:\4\vcpkg\buildtrees\simdutf\src\v3.2.15-c99b139426.clean\tools\sutf.cpp(138): note: see reference to function template instantiation 'void CommandLine::run_simdutf_procedure<CommandLine::run_procedure::<lambda_a55ba68c4042f8e6132c03ad63c272c3>>(PROCEDURE)' being compiled
>         with
>         [
>             PROCEDURE=CommandLine::run_procedure::<lambda_a55ba68c4042f8e6132c03ad63c272c3>
>         ]
> E:\4\vcpkg\buildtrees\simdutf\src\v3.2.15-c99b139426.clean\tools\sutf.cpp(276): warning C4018: '<': signed/unsigned mismatch
> ```
> 
> details： [install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/12289236/install-x64-windows-dbg-out.log)

This failure seems to occur because the simdutf library itself propagates the usage of the `/WX` flag to its consumers, which causes the warning in the tool to turn into an error. I have opened two separate PRs upstream: One to disable the propagation of warning flags (https://github.com/simdutf/simdutf/pull/266), and one to fix the code that issues the warning (https://github.com/simdutf/simdutf/pull/267). Either of those should fix this error.

If the simdutf maintainers accept my changes within the next few days I plan on simply updating the port to that new version. If I don't get an answer from them, I will add my changes as patches to the port.

Does that sound ok?

**Comment by JuliusBrueggemann** [comment]
The build failure on `x86-windows` has been fixed upstream with version 3.2.16. I've updated the port accordingly to use the new version.

**Review by Adela0814 — APPROVED** [review]

**Comment by Adela0814** [comment]
All feature test pass on the following triplets:

- x64-windows
- x64-windows-static
- x86-windows

Usage test pass on `x64-windows`.

**Pull request merged**